### ISSUE TITLE
CalendarConfig: "days" warning is unavoidable when using "dates"

### ISSUE THREAD

**Issue by DreadPirateShawn** [sakhnevych/calendar-view#30]
When passing "dates" to CalendarConfig, a warning will always be printed that "Both parameters 'days' and 'dates' are used. 'days' value will be skipped."

## Demo

```
$ cat demo.py 
#!/opt/homebrew/bin/python3
from calendar_view.core import data
from calendar_view.core.config import CalendarConfig
config = CalendarConfig(
    dates='Mo - Fr',
)
data.validate_config(config)
```
```
$ ./demo.py 
Both parameters 'days' and 'dates' are used. 'days' value will be skipped.
```

## Failed workaround

```
$ cat demo.py 
#!/opt/homebrew/bin/python3
from calendar_view.core import data
from calendar_view.core.config import CalendarConfig
config = CalendarConfig(
    days=None,
    dates='Mo - Fr',
)
data.validate_config(config)
```
```
$ ./demo.py 
Traceback (most recent call last):
  File "/Users/shawn.falkner-horine/Documents/github/joyent-misc/scripts/./demo.py", line 8, in <module>
    data.validate_config(config)
  File "/opt/homebrew/lib/python3.12/site-packages/calendar_view/core/data.py", line 17, in validate_config
    config.validate()
  File "/opt/homebrew/lib/python3.12/site-packages/calendar_view/core/config.py", line 64, in validate
    if not (0 < self.days <= 14):
            ^^^^^^^^^^^^^^^^^^^
TypeError: '<' not supported between instances of 'int' and 'NoneType'
```

## Root cause

The "days" parameter is defaulted to 7 here: https://github.com/sakhnevych/calendar-view/blob/adbb076d77f9b8f4c2a02f5a2b330240b4746cef/calendar_view/core/config.py#L25

and no allowance for None value is provided here: https://github.com/sakhnevych/calendar-view/blob/adbb076d77f9b8f4c2a02f5a2b330240b4746cef/calendar_view/core/config.py#L64

so therefore this warning is unavoidable: https://github.com/sakhnevych/calendar-view/blob/adbb076d77f9b8f4c2a02f5a2b330240b4746cef/calendar_view/core/config.py#L68

**Comment by DreadPirateShawn** [comment]
## Successful (albeit hacky) workaround

```
$ cat demo.py 
#!/opt/homebrew/bin/python3
import logging
class HackFilter(logging.Filter):
    def filter(self, record):
        return not record.getMessage().startswith("Both parameters 'days' and 'dates' are used.")
logger = logging.getLogger('calendar_view.core.config')
logger.addFilter(HackFilter())
from calendar_view.core import data
from calendar_view.core.config import CalendarConfig
config = CalendarConfig(
    dates='Mo - Fr',
    show_year=True,
    show_date=False,
)
data.validate_config(config)
```
```
$ ./demo.py 
'show_year' is set to True, but date wont be displayed, because 'show_date' is False.
```
(Note that the second warning is still logged, but not the "day/dates" warning -- this allows surgical quieting of this one warning, without discarding ALL warnings.)

**Comment by sakhnevych** [comment]
Hi @DreadPirateShawn. Thank you for raising the issue!

It is fixed in the `2.4.4` release.
